clippy_lints/src/operators/arithmetic.rs

```diff
@@ -50,36 +52,82 @@ impl Arithmetic {
         span_lint(cx, ARITHMETIC, expr.span, "arithmetic detected");
         self.expr_span = Some(expr.span);
     }
+
+    /// Checks assign operators (+=, -=, *=, /=) already handled by the CTFE.
+    fn has_valid_assign_op(op: &Spanned<hir::BinOpKind>, rhs: &hir::Expr<'_>) -> bool {
+        match op.node {
+            hir::BinOpKind::Div | hir::BinOpKind::Rem => match rhs.kind {
+                hir::ExprKind::Lit(_) => true,
+                hir::ExprKind::Unary(hir::UnOp::Neg, expr) => {
+                    let hir::ExprKind::Lit(lit) = &expr.kind else {
+                        return true;
```

Comment:
Could be `i /= -(something that ends up as 1);`

```suggestion
                        return false;
```